File: credit_check.py
from functools import reduce
def greater_than_nine(num):
    result = [int(x) for x in str(num)]
    sum = reduce((lambda x,y: x+y),result)
    return sum

def credit_check(str):
    step1 = []
    step2 = []
    new_str = str[::-1]
    index = 0

    for i in new_str:
        if index % 2 == 0:
            step1.append(int(i))
            index+=1
        elif index % 2 != 0:
            step1.append(int(i)*2)
            index+=1

    for i in step1:
        if i < 10:
            step2.append(i)
        else: step2.append(greater_than_nine(i))

    step3 = reduce((lambda x,y:x+y), step2)
    
    if step3 % 10 == 0:
        return 'The number is valid!'
    else:
        return 'The number is invalid!'

File: creditCheck.js
//'5541808923795240' => "The number is valid!"
exports.creditCheck = function(str) {
    let num = str.split('').reverse().map(Number)
    let num2 = []
    let num3 = []
    let product
    for (let n in num) {
        if (n % 2 === 0) {
            num2.push(num[n])
        } else {
            num2.push(num[n]*2)
        }
    }
    for (let n of num2) {
        if (n > 9) {
            digits = n.toString().split('').map(Number)
            product = digits.reduce((x,y) => x+y)
            num3.push(product)
        } else {
            num3.push(n)
        }
    }
    return num3.reduce((x,y) => x+y) % 10 === 0 ? 'The number is valid!' : 'The number is invalid!'
    
}


